### PR TITLE
b/157896907: More stats for Backend Auth, handle rejections properly.

### PR DESCRIPTION
**Pull request by nareddyt**
Very similar to #164, but for Backend Auth.
- Reject request when filter state is missing.
- Increment counters for different code paths.
- Test and document stats.

Signed-off-by: Teju Nareddy <nareddyt@google.com>

### CODE REVIEW

File: src/envoy/http/backend_auth/filter_test.cc [hunk 1]
```diff
@@ -86,9 +105,16 @@ TEST_F(BackendAuthFilterTest, NotHaveAudience) {
...
+          scope_, "backend_auth.allowed_by_no_configured_rules");
+  ASSERT_NE(counter, nullptr);
+  EXPECT_EQ(counter->value(), 1);
 }
 
-TEST_F(BackendAuthFilterTest, HasAudienceButGetsEmptyToken) {
+TEST_F(BackendAuthFilterTest, HasAudienceButGetsEmptyTokenRejected) {
   Envoy::Http::TestRequestHeaderMapImpl headers{{":method", "GET"},
                                                 {":path", "/books/1"}};
   utils::setStringFilterState(
```
Comment (TAOXUY) [inline src/envoy/http/backend_auth/filter_test.cc:120]: Can we reuse these lines of code by putting into function?

File: src/envoy/http/backend_auth/filter.cc [hunk 2]
```diff
@@ -38,6 +38,7 @@ constexpr char kBearer[] = "Bearer ";
 struct RcDetailsValues {
   // The request is rejected due to missing backend auth token.
   const std::string MissingBackendToken = "missing_backend_token";
+  const std::string MissingOperation = "missing_operation";
```
[the remark this answers is not in the document]
Comment (nareddyt) [reply src/envoy/http/backend_auth/filter.cc:43]: Done

File: src/envoy/http/backend_auth/filter_test.cc (hunk 1 above)
Comment (nareddyt) [reply src/envoy/http/backend_auth/filter_test.cc:120]: Done

**Pull request merged**